src/types/onyx/ReportViolation.ts

```diff
@@ -0,0 +1,16 @@
+import type {ValueOf} from 'type-fest';
+import type CONST from '@src/CONST';
+
+/**
+ * Names of violations.
+ * Derived from `CONST.VIOLATIONS` to maintain a single source of truth.
+ */
+type ReportViolationName = ValueOf<typeof CONST.REPORT_VIOLATIONS>;
+
+/**
+ * Report Violation model
+ */
+type ReportViolations = Record<ReportViolationName, Record<string, Record<string, unknown>>>;
```

Comment:
How about a slight change?
```suggestion
import type {EmptyObject} from 'type-fest';

type ReportFieldsViolations = Record<string, EmptyObject>;
type ReportViolations = Record<ReportViolationName, FieldsViolations>;
```

Reply:
Thanks!